lisa/tools/kdump.py

```diff
@@ -0,0 +1,326 @@
+    is not performed during a kexec boot. This has the effect of reducing the
+    time required for a reboot.
+    """
+
+    @property
+    def command(self) -> str:
+        return KernelDumpTool(self.node.os).get_tool_cmd_name("kexec")
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        tool_name = KernelDumpTool(self.node.os).get_tool_package_name("kexec")
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages(tool_name)
+        return self._check_exists()
+
+
+class Makedumpfile(Tool):
+    """
+    makedumpfile - make a small dumpfile of kdump
+    With kdump, the memory image of the first kernel can be taken as vmcore
+    while the second kernel is running. makedumpfile makes a small DUMPFILE by
+    compressing dump data or by excluding unnecessary pages for analysis, or both.
+    """
+
+    @property
+    def command(self) -> str:
+        return KernelDumpTool(self.node.os).get_tool_cmd_name("makedumpfile")
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        tool_name = KernelDumpTool(self.node.os).get_tool_package_name("makedumpfile")
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages(tool_name)
+        return self._check_exists()
+
+
+class Kdump(Tool):
+    """
+    kdump is a feature of the Linux kernel that creates crash dumps in the event of a
+    kernel crash. When triggered, kdump exports a memory image (also known as vmcore)
+    that can be analyzed for the purposes of debugging and determining the cause of a
+    crash.
+
+    kdump tool manages the kdump feature of the Linux kernel. Different distro os has
+    different kdump tool.
+
+    We can support Redhat, Suse, Debian family distro now. We will add support for
+    other distros.
+    """
+
+    # If the file /sys/kernel/kexec_crash_loaded does not exist. This means that the
+    # currently running kernel either was not configured to support kdump, or that a
+    # crashkernel= commandline parameter was not used when the currently running kernel
+    # booted. Value "1" means crash kernel is loaded, otherwise not loaded.
+    #
+    # It also has /sys/kernel/kexec_crash_size file, which record the crash kernel size
+    # of memory reserved. We don't need to check this file in our test case.
+    kexec_crash = "/sys/kernel/kexec_crash_loaded"
+
+    # This file shows you the current map of the system's memory for each physical
+    # device. We can check /proc/iomem file for memory reserved for crash kernel.
+    iomem = "/proc/iomem"
+
+    # Following are the configuration setting required for system and dump-capture
+    # kernels for enabling kdump support.
+    required_kernel_config = [
+        "CONFIG_KEXEC",
+        "CONFIG_CRASH_DUMP",
+        "CONFIG_PROC_VMCORE",
+    ]
+
+    dump_path = "/var/crash"
+    crashkernel_cfg_file = ""
```

Comment:
It's used in only one method, it doesn't need to be a private member in the class.

Reply:
Done